endgame_simulations/simulations.py

```diff
@@ -0,0 +1,232 @@
+from abc import ABC, abstractproperty
+from typing import ClassVar, Generic, Iterator, TypeVar, cast, overload
+
+import tqdm
+from hdf5_dataclass import FileType
+
+from endgame_simulations.models import BaseInitialParams
+
+from .common import AdvanceState, BaseState, State
+
+ParamsModel = TypeVar("ParamsModel", bound=BaseInitialParams)
+
+
+class GenericSimulation(Generic[ParamsModel, State], ABC):
+    state_class: ClassVar[type[BaseState]]
```

Comment:
Type?

Reply:
This is actually deprecated I learned today https://docs.python.org/3/library/typing.html#typing.Type